CryptoPkg/Library/BaseCryptLibOnOneCrypto/Readme.md

```diff
@@ -0,0 +1,925 @@
+---
+config:
+  layout: elk
+---
+classDiagram
+
+    namespace Protocol {
+        class OneCrypto:::protocol {
+            <<Protocol Interface>>
+            UINT16 Major;
+            UINT16 Minor;
+            // v1.0 ------------------------------------------
+            ONE_CRYPTO_PROVIDED_FUNC ONE_CRYPTO_PROVIDED_FUNC;
+            // ...
+        }
+    }
+
+    namespace Library {
+        class BaseCryptLib:::library {
+        }
+
+        class TlsLib:::library {
+
+        }
+
+        class HashApiLib:::library {
+
+        }
+    }
+
+    namespace LibraryInstance {
+        class BaseCryptOnOneCrypto:::library_instance {
+            <<Library Interface>>
+            EFI_STATUS CryptoFunc(..)
+        }
+    }
+
+    namespace Driver {
+        class OneCryptoBinSupvMm:::driver {
+            <<Driver>>
+            Phase Agnostic Crypto Driver
+            Shared across Phases
+
+            Crypto Entry(..) // phase agnostic code
+            SupvMm Entry(..) // stub entry
+        }
+
+        class OneCryptoLoaderSupvMm:::driver {
+            <<Driver>>
+        }
+
+        class OneCryptoLoaderDxe:::driver {
+            <<Driver>>
+        }
+    }
+
+    namespace GenericModules {
+        class GenericDriver:::module {
+            <<Driver>>
+            Any driver that needs crypto
+        }
+
+        class GenericLibrary:::module {
+            <<Library Interface>>
+            Any library that needs crypto
+        }
+
+    }
+
+    OneCryptoLoaderSupvMm --> OneCryptoBinSupvMm : Loads Phase Agnostic Code
+    OneCryptoLoaderDxe --> OneCryptoBinSupvMm : Loads Phase Agnostic Code
+
+    OneCryptoBinSupvMm ..> OneCryptoLoaderSupvMm : Provides Crypto
+    OneCryptoBinSupvMm ..> OneCryptoLoaderDxe : Provides Crypto
+
+    OneCryptoLoaderSupvMm ..|> OneCrypto : Publishes
+    OneCryptoLoaderDxe ..|> OneCrypto : Publishes
+
+    BaseCryptOnOneCrypto ..|> OneCrypto : Locates
+
+    BaseCryptOnOneCrypto <.. BaseCryptLib : Backed By
+    BaseCryptOnOneCrypto <.. TlsLib : Backed By
+
+
+    GenericLibrary ..> BaseCryptLib : Uses
+    GenericDriver ..> BaseCryptLib : Uses
+    GenericLibrary ..> TlsLib : Uses
+    GenericDriver ..> HashApiLib : Uses
+
+    classDef driver fill:#187286,stroke:#333,stroke-width:2px
+    classDef protocol fill:#d63230,stroke:#333,stroke-width:2px
+    classDef library fill:#1c77c3,stroke:#333,stroke-width:2px
+    classDef module fill:#ff8c00,stroke:#333,stroke-width:2px
+
+```
+
+**Architecture Components:**
+
+- **Platform Code**: Uses standard BaseCryptLib/TlsLib/HashApiLib APIs
+- **OneCrypto Loaders**: Contain enough logic to find and load the phase agnostic code
+- **OneCrypto Binary Driver**: Contains Phase agnostic cryptographic implementations
+- **BaseCryptLibOnOneCrypto Library**: Provides version checking, type safety, error handling, and API translation
+
+## How to Use BaseCryptLibOnOneCrypto
+
+### In Your Module's INF File
+
+Add BaseCryptLib to your LibraryClasses section:

```

Comment:
This is not true. You do not specify the library implementation in the module's INF file. You specify it in your platform's DSC file. Either in the [LibraryClasses] section or in the library class override of a specific component in the [Components] section.